### PR TITLE
Package satyrographos.0.0.2.11

### PR DESCRIPTION
**Pull request by na4zagin3**
### `satyrographos.0.0.2.11`
A package manager for SATySFi
Satyrographos is a package manager for [SATySFi].

Satyrographos is distributed under the LGPL-3.0 license.

  [SATySFi]: https://github.com/gfngfn/SATySFi
  [Satyrographos]: https://github.com/na4zagin3/satyrographos

---
* Homepage: https://github.com/na4zagin3/satyrographos
* Source repo: git+https://github.com/na4zagin3/satyrographos.git
* Bug tracker: https://github.com/na4zagin3/satyrographos/issues

---
:camel: Pull-request generated by opam-publish v2.1.0

### CODE REVIEW

File: packages/satyrographos/satyrographos.0.0.2.11/opam [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  "uri-sexp" {>= "3.0.0"}
+  "yaml" {>= "2.0" & < "3.0"}
+  "yojson"
+
+  # Janestreet Libs
+  "core" {>= "v0.14" & < "v0.15"}
+  "ppx_jane"
+  "shexp"
+]
+synopsis: "A package manager for SATySFi"
```
Comment (mseri) [inline packages/satyrographos/satyrographos.0.0.2.11/opam:44]:
```suggestion
conflicts: [ "result" {< "1.5"} ]
synopsis: "A package manager for SATySFi"
```

For
```
# (cd _build/default && /home/opam/.opam/4.09/bin/ocamlc.opt -w -40 -g -ppx '.ppx/b4175a47579235f9b841d5d1f64bf609/ppx.exe --as-ppx --cookie '\''library-name="satyrographos"'\''' -bin-annot -I src/.satyrographos.objs/byte -I /home/opam/.opam/4.09/lib/angstrom -I /home/opam/.opam/4.09/lib/base -I /home/opam/.opam/4.09/lib/base/base_internalhash_types -I /home/opam/.opam/4.09/lib/base/caml -I /home/opam/.opam/4.09/lib/base/md5 -I /home/opam/.opam/4.09/lib/base/shadow_stdlib -I /home/opam/.opam/4.09/lib/base_bigstring -I /home/opam/.opam/4.09/lib/base_quickcheck -I /home/opam/.opam/4.09/lib/bigstringaf -I /home/opam/.opam/4.09/lib/bin_prot -I /home/opam/.opam/4.09/lib/bin_prot/shape -I /home/opam/.opam/4.09/lib/biniou -I /home/opam/.opam/4.09/lib/bytes -I /home/opam/.opam/4.09/lib/core -I /home/opam/.opam/4.09/lib/core/error_checking_mutex -I /home/opam/.opam/4.09/lib/core_kernel -I /home/opam/.opam/4.09/lib/core_kernel/base_for_tests -I /home/opam/.opam/4.09/lib/core_kernel/caml_unix -I /home/opam/.opam/4.09/lib/core_kernel/flags -I /home/opam/.opam/4.09/lib/core_kernel/version_util -I /home/opam/.opam/4.09/lib/easy-format -I /home/opam/.opam/4.09/lib/fieldslib -I /home/opam/.opam/4.09/lib/fileutils -I /home/opam/.opam/4.09/lib/jane-street-headers -I /home/opam/.opam/4.09/lib/json-derivers -I /home/opam/.opam/4.09/lib/ocaml/threads -I /home/opam/.opam/4.09/lib/ocamlgraph -I /home/opam/.opam/4.09/lib/opam-core -I /home/opam/.opam/4.09/lib/opam-file-format -I /home/opam/.opam/4.09/lib/opam-format -I /home/opam/.opam/4.09/lib/parsexp -I /home/opam/.opam/4.09/lib/posixat -I /home/opam/.opam/4.09/lib/ppx_assert/runtime-lib -I /home/opam/.opam/4.09/lib/ppx_bench/runtime-lib -I /home/opam/.opam/4.09/lib/ppx_compare/runtime-lib -I /home/opam/.opam/4.09/lib/ppx_deriving/runtime -I /home/opam/.opam/4.09/lib/ppx_enumerate/runtime-lib -I /home/opam/.opam/4.09/lib/ppx_expect/collector -I /home/opam/.opam/4.09/lib/ppx_expect/common -I /home/opam/.opam/4.09/lib/ppx_expect/config -I /home/opam/.opam/4.09/lib/ppx_expect/config_types -I /home/opam/.opam/4.09/lib/ppx_hash/runtime-lib -I /home/opam/.opam/4.09/lib/ppx_here/runtime-lib -I /home/opam/.opam/4.09/lib/ppx_inline_test/config -I /home/opam/.opam/4.09/lib/ppx_inline_test/runtime-lib -I /home/opam/.opam/4.09/lib/ppx_module_timer/runtime -I /home/opam/.opam/4.09/lib/ppx_sexp_conv/runtime-lib -I /home/opam/.opam/4.09/lib/re -I /home/opam/.opam/4.09/lib/result -I /home/opam/.opam/4.09/lib/seq -I /home/opam/.opam/4.09/lib/sexplib -I /home/opam/.opam/4.09/lib/sexplib/unix -I /home/opam/.opam/4.09/lib/sexplib0 -I /home/opam/.opam/4.09/lib/shexp/bigstring -I /home/opam/.opam/4.09/lib/shexp/bigstring-io -I /home/opam/.opam/4.09/lib/shexp/process -I /home/opam/.opam/4.09/lib/shexp/sexp -I /home/opam/.opam/4.09/lib/spawn -I /home/opam/.opam/4.09/lib/splittable_random -I /home/opam/.opam/4.09/lib/stdio -I /home/opam/.opam/4.09/lib/stringext -I /home/opam/.opam/4.09/lib/time_now -I /home/opam/.opam/4.09/lib/timezone -I /home/opam/.opam/4.09/lib/typerep -I /home/opam/.opam/4.09/lib/uri -I /home/opam/.opam/4.09/lib/uri-sexp -I /home/opam/.opam/4.09/lib/variantslib -I /home/opam/.opam/4.09/lib/yojson -no-alias-deps -open Satyrographos -o src/.satyrographos.objs/byte/satyrographos__Library.cmi -c -intf src/library.mli)
# File "src/library.mli", line 110, characters 80-88:
# 110 | val read_dir_result : outf:Format.formatter -> FilePath.filename -> (t, string) Result.t
#                                                                                       ^^^^^^^^
# Error: Unbound type constructor Result.t
```

File: packages/satyrographos/satyrographos.0.0.2.11/opam [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+]
+
+depends: [
+  "ocaml" {>= "4.09.0"}
+
+  "conf-diffutils" {with-test}
+  "dune" {>= "2.7"}
+  "fileutils"
+  "json-derivers"
+  "menhir"
```
Comment (mseri) [inline packages/satyrographos/satyrographos.0.0.2.11/opam:26]:
```suggestion
  "menhir" {>= "20180523"}
```
Uses `--infer-write-query`

Comment (na4zagin3) [reply packages/satyrographos/satyrographos.0.0.2.11/opam]: Thank you!
Do you mind if I ask what command the option is for?

Comment (mseri) [reply packages/satyrographos/satyrographos.0.0.2.11/opam]: It is used when invoking menhir

Comment (na4zagin3) [reply packages/satyrographos/satyrographos.0.0.2.11/opam]: Got it.  Thank you for telling me that!

File: packages/satyrographos/satyrographos.0.0.2.11/opam [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  "dune" {>= "2.7"}
+  "fileutils"
+  "json-derivers"
+  "menhir" {>= "20180523"}
+  "ppx_deriving"
+  "ppx_deriving_yojson"
+  "ocamlgraph"
+  "opam-format" {>= "2.0" & < "2.1"}
+  "opam-state" {>= "2.0" & < "2.1"}
+  "re"
```
Comment (mseri) [inline packages/satyrographos/satyrographos.0.0.2.11/opam:32]:
```suggestion
  "re" { >= "1.9.0" }
```

Uses Re.Seq

**Comment by mseri** [comment]
Thanks @na4zagin3 
Please upstream the fixes

**Comment by na4zagin3** [comment]
Sure!  Thanks!

**Pull request merged**